debezium-server-core/src/main/java/io/debezium/server/DefaultChangeConsumer.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.server;
+
+import static io.debezium.openlineage.dataset.DatasetMetadata.STREAM_DATASET_TYPE;
+import static io.debezium.openlineage.dataset.DatasetMetadata.DataStore.KAFKA;
+import static io.debezium.openlineage.dataset.DatasetMetadata.DatasetKind.OUTPUT;
+
+import java.util.List;
+
+import org.apache.kafka.connect.header.ConnectHeaders;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.eclipse.microprofile.config.Config;
+
+import io.debezium.connector.common.DebeziumTaskState;
+import io.debezium.embedded.EmbeddedEngineChangeEvent;
+import io.debezium.engine.ChangeEvent;
+import io.debezium.engine.DebeziumEngine;
+import io.debezium.openlineage.ConnectorContext;
+import io.debezium.openlineage.DebeziumOpenLineageEmitter;
+import io.debezium.openlineage.dataset.DatasetDataExtractor;
+import io.debezium.openlineage.dataset.DatasetMetadata;
+
+public class DefaultChangeConsumer extends BaseChangeConsumer implements DebeziumEngine.ChangeConsumer<ChangeEvent<Object, Object>> {
+
+    private final DebeziumEngine.ChangeConsumer<ChangeEvent<Object, Object>> delegateConsumer;
+    private final Config config;
+    private final DatasetDataExtractor datasetDataExtractor;
+
+    public DefaultChangeConsumer(DebeziumEngine.ChangeConsumer<ChangeEvent<Object, Object>> delegateConsumer, Config config) {
+        this.delegateConsumer = delegateConsumer;
+        this.config = config;
+        datasetDataExtractor = new DatasetDataExtractor();
+    }
+
+    @Override
+    public void handleBatch(List<ChangeEvent<Object, Object>> records, DebeziumEngine.RecordCommitter<ChangeEvent<Object, Object>> committer)
+            throws InterruptedException {
+
+        final String sink = config.getValue("debezium.sink.type", String.class);
+
+        try {
+            this.delegateConsumer.handleBatch(records, committer);
+        }
+        catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+
+        records.forEach(record -> {
```

Comment:
Should we always enable it? IMHO there should be at least option to turn it off, ideally turn off by default and enabled only when user request it.

Reply:
But we still do this additional data processing, which is not needed at all and slows down the things. I believe we should skip whole this loop if the lineage is not turned on.